src/renderer/components/interact/forms/Bond.tsx

```diff
@@ -40,21 +42,20 @@ export const Bond: React.FC<Props> = ({ onFinish: onFinishProp, max, isLoading =
   )
 
   const amountValidator = useCallback(
-    (_, value: string, cb: (error?: string) => void) => {
-      const numberValue = Number(value)
-      if (numberValue <= 0 || Number.isNaN(numberValue)) {
-        cb(intl.formatMessage({ id: 'wallet.validations.graterThen' }, { value: 0 }))
-      }
-      if (max.amount().isLessThan(Number(value))) {
-        cb(
-          intl.formatMessage(
+    (_, value: string) => {
+      return validateTxAmountInput({
+        input: bnOrZero(value),
+        maxAmount: max,
+        errors: {
+          // We have InputBigNumber beneath and it allows input only of BN
+          msg1: '',
```

Comment:
Don't use if `validateTxAmountInput` if it does not fit your needs. Create your "own" validation by just using helpers from `helpers/form/validation` you really need.